tensorflow_io/core/kernels/audio_kernels.h

```diff
@@ -45,5 +54,40 @@ Status MP4ReadableResourceInit(
     Env* env, const string& input,
     std::unique_ptr<AudioReadableResourceBase>& resource);
 
+// Container for decoded audio.
+class DecodedAudio {
+public:
+  const bool success;
+  const int channels;
+  const int samples_perchannel;
+  const int sampling_rate;
+  // should first contain all samples of the left channel
+  // followed by the right channel
+  const int16 *data;
```

Comment:
I think we could avoid allocate the memory here, as we can create the output `Tensor` which will hold the memory. Then the output Tensor can be used directly to get the data?

Reply:
I was thinking about this for a while, but I'm not sure how it would work. The problem is I need a shape to allocate the output tensor, which I get by decoding the MP3, for which in turn I need to have memory.

Reply:
In case of a shape, in:
https://github.com/tensorflow/io/blob/7a19d3461a07929b4cad67734ccfba79d297674f/tensorflow_io/core/kernels/audio_mp3_kernels.cc#L103

a callback-type of lambda is passed which allows the allocation to be done when shape is ready. This woucl be helpful when we only want to call once to `read`.